play-route/src/main/scala/org/http4s/server/play/PlayRouteBuilder.scala

```diff
@@ -0,0 +1,149 @@
+package org.http4s.server.play
+
+import akka.stream.scaladsl.{Sink, Source}
+import akka.util.ByteString
+import cats.data.OptionT
+import cats.effect.{Effect, IO}
+import fs2.interop.reactivestreams._
+import org.http4s.server.play.PlayRouteBuilder.{PlayAccumulator, PlayRouting, PlayTargetStream}
+import org.http4s.{EmptyBody, Header, Headers, HttpService, Method, Request, Response, Uri}
+import play.api.http.HttpEntity.Streamed
+import play.api.libs.streams.Accumulator
+import play.api.mvc._
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, ExecutionContext, Future, Promise}
+
+class PlayRouteBuilder[F[_]](
+    service: HttpService[F]
+)(implicit F: Effect[F], executionContext: ExecutionContext) {
+
+  type UnwrappedKleisli = Request[F] => OptionT[F, Response[F]]
+  private[this] val unwrappedRun: UnwrappedKleisli = service.run
+
+  def requestHeaderToRequest(requestHeader: RequestHeader): Request[F] =
+    Request(
+      // todo fix the ???
+      method = Method.fromString(requestHeader.method).getOrElse(???),
+      uri = Uri(path = requestHeader.uri),
+      headers = Headers.apply(requestHeader.headers.toMap.toList.flatMap {
+        case (headerName, values) =>
+          values.map { value =>
+            Header(headerName, value)
+          }
+      }),
+      body = EmptyBody
+    )
+
+  type ResponseStream = fs2.Stream[F, Byte]
+
+  def convertStream(responseStream: ResponseStream): PlayTargetStream = {
+    val entityBody: fs2.Stream[F, Byte] = responseStream
+    Source
+      .fromPublisher(entityBody.toUnicastPublisher())
+      .map(byte => ByteString(byte))
```

Comment:
This is going to be a stream of single -byte `ByteString`s.  You might want to call `stream.chunks.map(chunk => ByteString(chunk.toArray))` or similar to preserve chunkiness from one stream library to the other.